src/EventStateMachineFactory.php

```diff
@@ -142,43 +162,113 @@ class EventStateMachineFactory extends AbstractBaseCallbackFactory implements St
+        $eventParams = null,
+        $eventTarget = null
+    ) {
+        $this->eventManager = $eventManager;
+        $this->eventFactory = $eventFactory;
+        $this->eventTarget  = $eventTarget;
+
+        $this->eventNameFormat = ($eventNameFormat !== null)
+            ? $this->_normalizeStringable($eventNameFormat)
+            : null;
+
+        $this->eventParams = ($eventParams !== null)
+            ? $this->_normalizeArray($eventParams)
+            : null;
     }
 
     /**
      * {@inheritdoc}
      *
      * @since [*next-version*]
      */
-    protected function _getFactoryCallback($config = null)
+    public function make($config = null)
     {
-        return function($config) {
-            $eventManager = $this->_containerGet($config, static::K_CFG_EVENT_MANAGER);
-            $eventFactory = $this->_containerGet($config, static::K_CFG_EVENT_FACTORY);
-            $initialState = $this->_containerGet($config, static::K_CFG_INITIAL_STATE);
-            $transitions  = $this->_containerGet($config, static::K_CFG_TRANSITIONS);
-
-            $eventNameFormat = $this->_containerHas($config, static::K_CFG_EVENT_NAME_FORMAT)
-                ? $this->_containerGet($config, static::K_CFG_EVENT_NAME_FORMAT)
-                : null;
-
-            $target = $this->_containerHas($config, static::K_CFG_EVENT_TARGET)
-                ? $this->_containerGet($config, static::K_CFG_EVENT_TARGET)
-                : null;
-
-            $params = $this->_containerHas($config, static::K_CFG_EVENT_PARAMS)
-                ? $this->_containerGet($config, static::K_CFG_EVENT_PARAMS)
-                : [];
-
-            return new EventStateMachine(
-                $eventManager, $eventFactory, $initialState, $transitions, $eventNameFormat, $target, $params
-            );
-        };
+        $initialState = $this->_containerGet($config, static::K_CFG_INITIAL_STATE);
+        $transitions  = $this->_containerGet($config, static::K_CFG_TRANSITIONS);
+
+        $eventManager = $this->_containerHas($config, static::K_CFG_EVENT_MANAGER)
+            ? $this->_containerGet($config, static::K_CFG_EVENT_MANAGER)
+            : $this->eventManager;
+            
+        $eventFactory = $this->_containerHas($config, static::K_CFG_EVENT_FACTORY)
+            ? $this->_containerGet($config, static::K_CFG_EVENT_FACTORY)
+            : $this->eventFactory;
+
+        $eventNameFormat = $this->_containerHas($config, static::K_CFG_EVENT_NAME_FORMAT)
+            ? $this->_containerGet($config, static::K_CFG_EVENT_NAME_FORMAT)
+            : $this->eventNameFormat;
+
+        $target = $this->_containerHas($config, static::K_CFG_EVENT_TARGET)
+            ? $this->_containerGet($config, static::K_CFG_EVENT_TARGET)
+            : $this->eventTarget;
+
+        $params = $this->_containerHas($config, static::K_CFG_EVENT_PARAMS)
+            ? $this->_containerGet($config, static::K_CFG_EVENT_PARAMS)
+            : $this->eventParams;
```

Comment:
I'd say: normalize params to array here; normalize the constructor arg [to iterable](https://github.com/Dhii/normalization-helper-base/blob/develop/src/NormalizeIterableCapableTrait.php) in constructor.